src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -709,26 +709,101 @@ public FieldTypeClass(int id, string name, int typeId, bool isNotPrivate, FieldA
-        public ValueTypeClass(string varName, byte[] buffer, JArray json, int id, bool expand_properties, int valueTypeId)
-        {
-            valueTypeBuffer = buffer;
-            valueTypeJson = json;
-            typeId = id;
-            valueTypeJsonProps = null;
-            valueTypeProxy = null;
-            valueTypeVarName = varName;
-            valueTypeAutoExpand = expand_properties;
+        private JArray Json { get; }
+        private int TypeId { get; }
+        private bool AutoExpand { get; }
+        private int Id { get; }
+        private JArray Proxy { get; set; }
+        private JArray JsonProps { get; set; }
+
+        public byte[] Buffer { get; }
+
+        public ValueTypeClass(byte[] buffer, JArray json, int id, bool expand_properties, int valueTypeId)
+        {
+            Buffer = buffer;
+            Json = json;
+            TypeId = id;
+            JsonProps = null;
+            Proxy = null;
+            AutoExpand = expand_properties;
             Id = valueTypeId;
         }
+
+        public async Task<JArray> GetProxy(MonoSDBHelper sdbAgent, CancellationToken token)
+        {
+            if (Proxy != null)
+                return Proxy;
+            Proxy = new JArray(Json);
+
+            var retDebuggerCmdReader = await sdbAgent.GetTypePropertiesReader(TypeId, token);
+            if (retDebuggerCmdReader == null)
+                return null;
+
+            var nProperties = retDebuggerCmdReader.ReadInt32();
+
+            for (int i = 0; i < nProperties; i++)
+            {
+                retDebuggerCmdReader.ReadInt32(); //propertyId
+                string propertyNameStr = retDebuggerCmdReader.ReadString();
+
+                var getMethodId = retDebuggerCmdReader.ReadInt32();
+                retDebuggerCmdReader.ReadInt32(); //setmethod
+                retDebuggerCmdReader.ReadInt32(); //attrs
+                if (await sdbAgent.MethodIsStatic(getMethodId, token))
+                    continue;
+                using var command_params_writer_to_proxy = new MonoBinaryWriter();
+                command_params_writer_to_proxy.Write(getMethodId);
+                command_params_writer_to_proxy.Write(Buffer);
+                command_params_writer_to_proxy.Write(0);
+                var (data, length) = command_params_writer_to_proxy.ToBase64();
+                Proxy.Add(JObject.FromObject(new
+                {
+                    get = JObject.FromObject(new
+                    {
+                        commandSet = CommandSet.Vm,
+                        command = CmdVM.InvokeMethod,
+                        buffer = data,
+                        length = length,
+                        id = MonoSDBHelper.GetNewId()
+                    }),
+                    name = propertyNameStr
+                }));
+            }
+            return Proxy;
+        }
+
+        public async Task<JArray> GetPropertiesValuesOfValueType(MonoSDBHelper sdbAgent, CancellationToken token)
```

Comment:
`OfValueType` can be dropped from the name, since it's a method on `ValueTypeClass` now.